control-plane/agents/src/bin/core/controller/scheduling/affinity_group.rs

```diff
@@ -12,32 +12,36 @@ pub(crate) fn get_restricted_nodes(
     affinity_group_spec: &AffinityGroupSpec,
     registry: &Registry,
 ) -> Vec<NodeId> {
-    let mut restricted_nodes: Vec<NodeId> = Vec::new();
-    if volume_spec.num_replicas == 1 {
-        let specs = registry.specs();
-        // Get the list of volumes being part of the Affinity Group.
-        let affinity_group_volumes = affinity_group_spec.volumes();
-        // Remove the current volume from the list, as it is under creation.
-        let affinity_group_volumes: Vec<_> = affinity_group_volumes
-            .iter()
-            .filter(|volume_id| *volume_id != volume_spec.uid())
-            .collect();
-        // List of restricted nodes, which already has replicas from the volumes of the
-        // Affinity Group.
-        for volume in &affinity_group_volumes {
-            // Fetch the list of nodes where the volume replicas are placed. If some
-            // volume doesn't exist yet, this list will be empty
-            // for that volume.
-            let node_ids = specs.volume_replica_nodes(volume);
-            // Add a new node in the list if it doesn't already exist.
-            let filtered_nodes: Vec<NodeId> = node_ids
-                .into_iter()
-                .filter(|id| !restricted_nodes.contains(id))
-                .collect();
-            restricted_nodes.extend(filtered_nodes);
-        }
+    // Since num_replicas in VolumeSpec is equal to 1 when scaling up from 1 replica to 2 replicas
```

Comment:
But SetReplica(2) can also happen when scaling down?

Reply:
This code path is not a part of the decrease volume replica flow.

Reply:
This only gets called from increase volume replicas

Reply:
I will extend my current to validate we are able to scale all the way up to 3 and then come back to 2, as we can't scale down beyond 2 at the moment as per initial design.

Reply:
Oh I see, fair enough then